.github/workflows/ci.yaml

```diff
@@ -162,61 +162,83 @@ jobs:
       run:
         working-directory: ./fe2-android
 
+    # This CI pipeline is greatly inspired by the sample given by https://github.com/ReactiveCircus/android-emulator-runner
     steps:
-    - name: Setup repo
-      uses: actions/checkout@v2
-      with:
-        submodules: recursive
-        fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of Sonar analysis
-
-    - name: Setup JDK 11
-      uses: actions/setup-java@v1
-      with:
-        java-version: 11.0.10
-
-    - name: Cache SonarCloud packages
-      uses: actions/cache@v1
-      with:
-        path: ~/.sonar/cache
-        key: ${{ runner.os }}-sonar
-        restore-keys: ${{ runner.os }}-sonar
-
-    - name: Cache Gradle packages
-      uses: actions/cache@v1
-      with:
-        path: ~/.gradle/caches
-        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
-        restore-keys: ${{ runner.os }}-gradle
-
-    - name: Grant execute permission for gradlew
-      run: |
-        chmod +x ./gradlew
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of Sonar analysis
+        
+      - name: Setup JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '11'  
 
-    - name: Build
-      shell: bash
-      run: |
-        # use grep to remove log spam due to jetifier transformation, c.f. https://stackoverflow.com/a/52353446
-        # use set -o pipefail to keep error code from gradle, c.f. https://stackoverflow.com/a/37128767
-        ( set -o pipefail; ./gradlew build --parallel --build-cache --info | grep -vE "^Transforming (.*) with (.*)Transform" )
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+
+      - name: Grant execute permission for gradlew
+        run: |
+          chmod +x ./gradlew
+
+      - name: Build
+        shell: bash
+        run: |
+          # use grep to remove log spam due to jetifier transformation, c.f. https://stackoverflow.com/a/52353446
+          # use set -o pipefail to keep error code from gradle, c.f. https://stackoverflow.com/a/37128767
+          ( set -o pipefail; ./gradlew build --parallel --build-cache --info | grep -vE "^Transforming (.*) with (.*)Transform" )
+
+      - name: Cache AVD
```

Comment:
Can we add some documentation on this AVD casing: at least why it is needed ? It doesn't seem to run in the check steps.

Reply:
`Cache AVD` is executed, but `Create AVD and generate snapshot for caching` is only executed the first time when there is no cache available to generate it. That is why it is skipped on later runs.